File: spring-web-java/src/main/java/spring/web/java/domain/model/entity/MemberPermission.java
package spring.web.java.domain.model.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class MemberPermission extends Base {

	@Id
	@GeneratedValue
	@Column(name = "member_permission_id")
	private Long id;

	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "member_id")
	private Member member;

	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "permission_id")
	private Permission permission;

	public static MemberPermission create(Permission permission) {
		MemberPermission memberPermission = new MemberPermission();

		memberPermission.permission = permission;

		return memberPermission;
	}

	protected void setMember(Member member) {
		this.member = member;
	}
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/domain/model/entity/MemberPermission.kt
package spring.web.kotlin.domain.model.entity

import jakarta.persistence.*

@Entity
class MemberPermission protected constructor(
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "permission_id", foreignKey = ForeignKey(ConstraintMode.NO_CONSTRAINT))
    var permission: Permission,
) {
    @Id
    @GeneratedValue
    @Column(name = "member_permission_id")
    var id: Long? = null
        protected set

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "member_id", foreignKey = ForeignKey(ConstraintMode.NO_CONSTRAINT))
    var member: Member? = null
        protected set

    companion object {
        fun create(permission: Permission) = MemberPermission(permission = permission)
    }

    fun associateMember(member: Member) {
        this.member = member
    }
}